script/DCA_v2.s.sol

```diff
@@ -0,0 +1,234 @@
+            _salt: 'MF-DCAHubPositionDescriptor-V1',
+            _args: hex""
+        });
+
+        // Chainlink Registry
+        address chainlinkFeedRegistry;
+        // On Ethereum, we use the `ChainlinkRegistry` operated by Chainlink.
+        if (block.chainid == 1) {
+            chainlinkFeedRegistry = 0x47Fb2585D2C56Fe188D0E6ec628a38b74fCeeeDf;
+        } else {
+            chainlinkFeedRegistry = deployCreate3({
+                _name: "ChainlinkRegistry",
+                _salt: 'MF-Chainlink-Feed-Registry-V1',
+                _args: abi.encode(msig, msigArray)
+            });
+        }
+
+        // Transformers
+        deployCreate3({_name: "ERC4626Transformer", _salt: 'MF-ERC4626-Transformer-V1', _args: abi.encode(msig)});
+        deployCreate3({_name: "ProtocolTokenWrapperTransformer", _salt: 'MF-Protocol-Transformer-V1', _args: abi.encode(msig)});
+        address transformerRegistry = deployCreate3({_name: "TransformerRegistry", _salt: 'MF-Transformer-Registry-V1', _args: abi.encode(msig)});
+        if (block.chainid == 1) {
+            address stETH = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+            deployCreate3({_name: "wstETHTransformer", _salt: 'MF-wstETH-Transformer-V1', _args: abi.encode(stETH, msig)});
+        }
+
+        // Oracles
+        address transformerOracle = deployOracles({
+            _transformerRegistry: transformerRegistry,
+            _chainlinkFeedRegistry: chainlinkFeedRegistry
+        });
+
+        // DCA V2 Core
+        address permissionsManager = deployCreate3({_name: 'DCAPermissionsManager', _salt: 'MF-DCAV2-PermissionsManager-V1', _args:  abi.encode(msig, tokenDescriptor)});
+        address timelock = deployCreate3({_name: "TimelockController", _salt: 'MF-DCAV2-Timelock-V1', _args: abi.encode(3 days, msigArray, msigArray)});
+        address dcaHub = deployCreate3({_name: "DCAHub", _salt: 'MF-DCAV2-DCAHub-V1', _args: abi.encode(msig, timelock, transformerOracle, permissionsManager)});
+
+        // Swappers
+        address swapperRegistry = deployCreate3({_name: "SwapperRegistry", _salt: 'MF-Swapper-Registry-V1', _args: abi.encode(new address[](0), new address[](0), msig, msigArray)});
+        deployCreate3({_name: 'SwapProxy', _salt: 'MF-Swap-Proxy-V1', _args: abi.encode(swapperRegistry, msig)});
+
+        // DCA V2 Periphery
+        deployCreate3({_name: 'DCAHubCompanion', _salt: 'MF-DCAV2-DCAHubCompanion-V5', _args: abi.encode(permit2AdapterAddress, permit2AdapterAddress, msig, PERMIT2)});
+        deployCreate3({_name: 'CallerOnlyDCAHubSwapper', _salt: 'MF-DCAV2-CallerDCAHubSwapper-V2', _args: hex""});
+        deployCreate3({_name: 'ThirdPartyDCAHubSwapper', _salt: 'MF-DCAV2-3PartySwapper-V3', _args: hex""});
+        deployCreate3({_name: 'DCAFeeManager', _salt: 'MF-DCAV2-DCAFeeManager-V3', _args: abi.encode(msig, msigArray)});
+        if (block.chainid == 1) {
+            address keep3r = 0xeb02addCfD8B773A5FFA6B9d1FE99c566f8c44CC;
```

Comment:
The `keep3r` address was hardcoded in the [original hardhat-deploy script](https://github.com/Mean-Finance/dca-v2-periphery/blob/main/deploy/005_keep3r_job.ts#L9), so I kept it the same here